src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/ProjectKNuGetProject.cs

```diff
@@ -76,18 +75,31 @@ public ProjectKNuGetProject(INuGetPackageManager project, string projectName, st
                 throw new ArgumentException(ProjectManagement.Strings.PackageStreamShouldBeSeekable);
             }
 
-            nuGetProjectContext.Log(ProjectManagement.MessageLevel.Info, Strings.InstallingPackage, packageIdentity);
+            nuGetProjectContext.Log(MessageLevel.Info, Strings.InstallingPackage, packageIdentity);
 
             packageStream.Seek(0, SeekOrigin.Begin);
-            var packageSupportedFrameworks = GetSupportedFrameworks(packageStream);
-            var projectFrameworks = _project.GetSupportedFrameworksAsync(token)
-                .Result
-                .Select(f => NuGetFramework.Parse(f.FullName));
+            
+            // Get additional information from the package that the INuGetPackageManger can act on.
+            IEnumerable<NuGetFramework> supportedFrameworks;
+            IEnumerable<PackageType> packageTypes;
+            using (var packageReader = new PackageArchiveReader(packageStream, leaveStreamOpen: true))
+            {
+                supportedFrameworks = packageReader.GetSupportedFrameworks();
+                packageTypes = packageReader.GetPackageTypes();
+            }
 
             var args = new Dictionary<string, object>();
-            args["Frameworks"] = projectFrameworks.Where(
-                projectFramework =>
-                    IsCompatible(projectFramework, packageSupportedFrameworks)).ToArray();
+
+            args["Frameworks"] = _project
+                .GetSupportedFrameworksAsync(token)
+                .Result
+                .Select(f => NuGetFramework.Parse(f.FullName))
+                .Where(projectFramework => IsCompatible(projectFramework, supportedFrameworks))
+                .ToArray();
+
+            args["PackageTypes"] = packageTypes
+                .ToArray();
```

Comment:
Changing `"Frameworks"` to `IEnumerable` would break contract. Existing functionality sends a `NuGetFramework[]`. Since this is calling into someone else's code, I would rather fully enumerate the sequence of package types (as we have already done with `"Frameworks"`).